combine_chunks.py

```diff
@@ -143,10 +160,7 @@ def combine_chunks(
 
             # Write these properties to the output file
             for name, size, unit, description in zip(names, sizes, units, descriptions):
-                phdf5.collective_write(
-                    outfile, name, data[name], create_dataset=False, comm=comm_world
-                )
-
+                lossy_filters.collective_write(outfile, name, data[name], comm_world)
             del data
 
     with MPITimer("Writing subhalo ranking by mass", comm_world):
```

Comment:
Should SubhaloRankByBoundMass also be written by lossy_filters?

Reply:
We can do that for consistency. lossy_filters.collective_write explicitly converts the data to the same type as the dataset in the file before passing it to HDF5, which helps when we're using odd types like reduced precision floats because HDF5 refuses to do collective I/O if data conversion is required. In this case the dataset just contains int32s so it doesn't really matter.